mmseg/datasets/samplers/distributed_sampler.py

```diff
@@ -0,0 +1,243 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from __future__ import division
+import math
+from typing import Iterator, Optional, Sequence
+
+import torch
+import torch.distributed as dist
+from torch.utils.data import Dataset
+from torch.utils.data import DistributedSampler as _DistributedSampler
+from torch.utils.data import Sampler
+
+from mmseg.core.utils import sync_random_seed
+
+
+class DistributedSampler(_DistributedSampler):
+    """DistributedSampler inheriting from
+    `torch.utils.data.DistributedSampler`.
+
+    In pytorch of lower versions, there is no `shuffle` argument. This child
+    class will port one to DistributedSampler.
```

Comment:
might remove these lines, `shuffle` has been supported from pytorch 1.2, and mmseg requires pytorch >= 1.5